python/cudf/cudf/core/frame.py

```diff
@@ -1557,14 +1557,18 @@ def _apply(self, func):
         """
         Apply `func` across the rows of the frame.
         """
-        kernel, retty = compile_or_get(self, func)
+        frame_metadata = _FrameJitMetadata(self)
+        kernel, retty = compile_or_get(frame_metadata, func)
 
         # Mask and data column preallocated
         ans_col = cupy.empty(len(self), dtype=retty)
         ans_mask = cudf.core.column.column_empty(len(self), dtype="bool")
         launch_args = [(ans_col, ans_mask)]
         offsets = []
-        for col in self._data.values():
+
+        # if compile_or_get succeeds, it is safe to create a kernel that only
+        # consumes the columns that are of supported dtype
+        for name, col in frame_metadata.supported_cols.items():
```

Comment:
The `name` isn't used.
```suggestion
        for col in frame_metadata.supported_cols.values():
```